src/ViewInstanceWrapper.js

```diff
@@ -1,26 +1,34 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { omit } from 'lodash';
 
 import { checkIfUserInMemberTenant } from '@folio/stripes/core';
 import { withTags } from '@folio/stripes/smart-components';
 
 import ViewInstance from './ViewInstance';
 import { useUserTenantPermissions } from './hooks';
 import { checkIfSharedInstance } from './utils';
+import {
+  useSearchInstanceByIdQuery,
+  useInstanceQuery,
+} from './common';
 
 const ViewInstanceWrapper = (props) => {
   const {
     match: { params: { id } },
-    resources,
     stripes,
   } = props;
 
-  const instance = resources.instance?.records?.[0];
-  const selectedInstance = instance?.id === id ? instance : null;
   const userId = stripes?.user?.user?.id;
+  const tenantId = stripes.okapi.tenant;
   const centralTenantId = stripes.user.user?.consortium?.centralTenantId;
 
+  const { instance: selectedInstance } = useSearchInstanceByIdQuery(id);
+  const { instance } = useInstanceQuery(
```

Comment:
Please extend the `checkIfSharedInstance` function, now we also need to check if `intance.shared`, because `source` will not always be with consortia prefix for member tenant.

Reply:
Hi @Dmytro-Melnyshyn to extend the `checkIfSharedInstance` function it's needed to make an additional call everywhere where it's used already and pass the `instance` property containing the `shared` prop, because now all the components that use this function are fetching instance by `inventory/instances/:{id}` endpoint, and here I used `search/instances/query=id=={id}`